src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverterBase.cs

```diff
@@ -313,9 +313,17 @@ public IndexFieldsMapping GetKnownFieldsForWriter()
                 break;
 
             case ValueType.DynamicJsonObject:
+                if (_index.Type.IsAuto())
+                {
+                    InsertRegularField(field, value.ToString(), indexContext, builder, sourceDocument, out shouldSkip);
```

Comment:
Given that we are interesting in existence, and that this is potentially expensive, what about if we index a constant value?

Reply:
Do you mean to support only `Field = null` in case of auto index and complex field?

Reply:
Anything else will not really work, right?

I don't think we'll see users doing something like `Where(u=>u.Address = new { City="London", Country = "UK"})`

Reply:
How likely is it that a user will do something like that?
For that matter, do we want to _allow_ that? It is almost accidental that this works for Lucene, and we have a chance to do this more nicely

Reply:
Hmm, I don't know. I wonder if writing some constant value as a marker for complex is not too misleading. I would rather say it's the user's responsibility to write this query in the correct way since we're sending an alert about it. If somebody wants to use it - why forbid it?

Reply:
Let's go the other way, then? Why are we storing this as a full string?
Can we store a hash? Would that make sense?

I would like to avoid dealing with large JSONs (remember, we are limited to 512 bytes anyway)

Reply:
What is benefit of storing a hash instead of constant value (e.g. `[JSON_VALUE]`)? 

The more I'm thinking about it the more I like the idea of storing something different than actual JSON value. It's a good point that we're limited to 512 bytes anyway. This means that most cases it will work, except big JSONs that will exceed this size. Then users might be surprised that an index doesn't return results for those entries.

If we store a different value (either a hash or a constant) then a user won't get expected results. Then it's more likely he'll look at the studio and see the alert. 

Queries checking if a value exist or is / isn't `null` will still work but indexing that info won't be expensive.

Reply:
You can still do equality check with a hash, as well as existence.
Range queries won't work, but aren't generally meaningful here either.
512 is actually pretty small and easy to hit. For example, here are 512 bytes on JSON:

```
{
  "street": "123 Main Street",
  "apartment_number": "Apt 4B",
  "building_name": "Sunrise Apartments",
  "suburb": "Downtown",
  "city": "Anytown",
  "state": "CA",
  "postal_code": "12345",
  "country": "United States",
  "latitude": 37.12345,
  "longitude": -122.67890,
  "phone": "+1 (555) 123-4567",
  "email": "john.doe@example.com",
  "additional_notes": "Near the city park and grocery store",
  "landmark": "City Tower",
  "region": "West Coast",
  "timezone": "Pacific Time (PT)"
}
```

And I think that I missed something, the point is that we _will_ hash the query value as well, so you could find it.

Reply:
Actually, we hash values that exceed 512 characters already:
https://github.com/ravendb/ravendb/blob/ee979067d822fb48bdace65d47fcc3e92335cd0c/src/Corax/IndexWriter.cs#L1289-L1309

Reply:
Yes, I want to extend that to normal JSON as well, smaller than 512

Reply:
We have a couple of issues with this approach:
- How to detect this is a JSON in querying? That means we would have to insert JSON fields into some tree as metadata and do a check on every query.
- What should happen when there are some items already (standard ones)? If we mark the field as JSON field that would result in the old values being unqueryable.

Reply:
We can do that from the client side, I think. By adding a query param for this.
But good point on the mixed mode scenario.